### PR TITLE
Adding `alibi` and `alibi-detect` from PyPI

### PR DESCRIPTION
**Pull request by sugatoray**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sugatoray** [comment]
## Build Failure: `pip check`

[(error)](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=443423&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1799)

```sh
import: 'alibi'
import: 'alibi'
+ pip check
tensorflow 2.4.3 has requirement absl-py~=0.10, but you have absl-py 1.0.0.
tensorflow 2.4.3 has requirement wrapt~=1.12.1, but you have wrapt 1.13.3.
Tests failed for alibi-0.6.3-py39hf3d152e_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
WARNING:conda_build.build:Tests failed for alibi-0.6.3-py39hf3d152e_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
WARNING conda_build.build:tests_failed(2970): Tests failed for alibi-0.6.3-py39hf3d152e_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
TESTS FAILED: alibi-0.6.3-py39hf3d152e_0.tar.bz2
```

**Comment by sugatoray** [comment]
## Reason for Windows Build Failure

[(error)](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=443424&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=06d62fd2-ab49-5add-9ee5-83b6882d7d2b&l=845)

```sh
The reported errors are:
- Encountered problems while solving:
-   - nothing provides requested tensorflow >=2.0.0,!=2.6.0,!=2.6.1,<2.8.0
-   - nothing provides pytorch needed by transformers-4.10.0-pyhd8ed1ab_0
```

**Comment by sugatoray** [comment]
PR is ready. Request for review/merging.

cc: @conda-forge/help-python

File: recipes/alibi/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+test:
+  imports:
+    - alibi
+  commands:
+    - pip check
+  requires:
+    - pip
+    # Necessary for passing pip-check
+    - absl-py >=0.10.0,<0.11.0
```
Comment (wolfv) [inline recipes/alibi/meta.yaml:47]: why are they not part of `run` if they are necessary to pass `pip check`?

Comment (sugatoray) [reply recipes/alibi/meta.yaml:47]: The build did not fail until the pip-check happened. So, am not clear on this part. Adding them to pip-requires, seems to make it pass without failure.

Comment (sugatoray) [reply recipes/alibi/meta.yaml]: Let me see what happens if I just move those under run-requirements.

**Comment by sugatoray** [comment]
## Reason for Build Failure

I added another recipe `alibi-detect`. This failure is because of that.

[(error)](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=443441&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=2252)

```sh
import: 'alibi_detect'
+ pip check
alibi-detect 0.8.1 requires opencv-python, which is not installed.
```

**Comment by sugatoray** [comment]
@wolfv Made changes. Please have a look at the recipe abd let me know if something else stands out. The pip-check for `alibi-dtect` was dropped as it looks for `opencv-python` while opencv has been installed already. So, I may have to remove some comments finally.

File: recipes/alibi/meta.yaml (hunk 1 above)
Comment (sugatoray) [reply recipes/alibi/meta.yaml]: Please have a look at the recipes now and let me know if you think anything else must be changed.

File: recipes/alibi/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/alibi-{{ version }}.tar.gz
+  sha256: 20131a369871b32bc7abfe1caa47131657abc0e41c7f5c8e0d2cc6105a816a96
+
+build:
+  number: 0
```
Comment (wolfv) [inline recipes/alibi/meta.yaml:13]: `noarch: python` ?

**Comment by wolfv** [comment]
Looks good to me, except for the noarch comment.

File: recipes/alibi/meta.yaml (hunk 2 above)
Comment (sugatoray) [reply recipes/alibi/meta.yaml:13]: Are you wondering if this could be made a noarch package?

Comment (sugatoray) [reply recipes/alibi/meta.yaml:13]: Okay. Just tested with `noarch: python` and it went fine. Please the results here:

- https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=444410&view=results

I had some lines commented temporarily. Will update/drop them now.

**Comment by sugatoray** [comment]
> Looks good to me, except for the noarch comment.

Now both recipes are `noarch: python`.

File: recipes/alibi/meta.yaml (hunk 2 above)
Comment (sugatoray) [reply recipes/alibi/meta.yaml:13]: Done.

**Comment by sugatoray** [comment]
Thank you, @beckermr.

**Pull request merged**